x-pack/plugins/encrypted_saved_objects/server/plugin.ts

```diff
@@ -48,18 +50,21 @@ export class Plugin {
     core: CoreSetup,
     deps: PluginsSetup
   ): Promise<EncryptedSavedObjectsPluginSetup> {
-    const {
-      config: { encryptionKey },
-      usingEphemeralEncryptionKey,
-    } = await createConfig$(this.initializerContext).pipe(first()).toPromise();
-
-    const crypto = nodeCrypto({ encryptionKey });
-
+    const config = await createConfig$(this.initializerContext).pipe(first()).toPromise();
     const auditLogger = new EncryptedSavedObjectsAuditLogger(
       deps.security?.audit.getLogger('encryptedSavedObjects')
     );
+
+    // This list of cryptos consists of the primary crypto that is used for both encryption and
+    // decryption, and the optional secondary cryptos that are used for decryption only.
+    const cryptos = [
```

Comment:
> I don't love this, but given its usage on the encryption service, it feels tolerable. I worry about consumers having to know that the first entry can be used for encryption, and all other keys can only be used for decryption.

I see, we can try to split this constructor argument into two separate ones and join inside constructor so that for consumers it will be something like `primaryCrypto` and optional `decryptionOnlyCryptos`?

> What do you think about creating a "patched" version of the "decryption only" crypto instances to throw an error if we ever attempt to call one of the encryption functions?

You mean wrap/patch them inside of the constructor? If so, yeah I can try to do that and we'll see how that looks like.

Reply:
> You mean wrap/patch them inside of the constructor? If so, yeah I can try to do that and we'll see how that looks like.

Yeah something like that is what I had in mind.

> I see, we can try to split this constructor argument into two separate ones and join inside constructor so that for consumers it will be something like primaryCrypto and optional decryptionOnlyCryptos?

I'd be happy with either of these approaches -- don't feel like you have to implement both. Whichever you feel is better is good with me 👍